apps/st2-history/controller.js

```diff
@@ -253,6 +256,46 @@ angular.module('main')
       }
     };
 
+    $scope.rerun = (function () {
+      var rerun = $scope.$new();
+
+      rerun.metaSpec = {
+        type: 'object',
+        properties: {
+          ref: {
+            type: 'string',
+            name: 'Action'
+          }
+        }
+      };
+
+
+      rerun.open = function () {
+        $scope.$root.state.go('^.rerun', {id: $scope.record.id});
+        rerun.payload = _.clone($scope.payload);
```

Comment:
So we clone all the parameters on the client side and send all of them to the server (or we just send the overrides)?

If we do the former I'm fine fine that since it probably keeps the client side code simpler (we don't need to figure what has changed and the end result should be the same anyway).